libs/eo/transfers/src/lib/eo-transfers-modal.component.ts

```diff
@@ -0,0 +1,136 @@
+  Input,
+  ViewChild,
+} from '@angular/core';
+import {
+  FormControl,
+  FormGroup,
+  FormsModule,
+  ReactiveFormsModule,
+  Validators,
+} from '@angular/forms';
+import { WattButtonComponent } from '@energinet-datahub/watt/button';
+import { WattDatepickerComponent, WattRange } from '@energinet-datahub/watt/datepicker';
+import { WATT_FORM_FIELD } from '@energinet-datahub/watt/form-field';
+import { WattInputDirective } from '@energinet-datahub/watt/input';
+import { WATT_MODAL, WattModalComponent } from '@energinet-datahub/watt/modal';
+import { WattRangeValidators } from '@energinet-datahub/watt/validators';
+import { EoTransfersService } from './eo-transfers.service';
+import { EoTransfersStore } from './eo-transfers.store';
+
+@Component({
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  selector: 'eo-transfers-modal',
+  imports: [
+    WATT_MODAL,
+    WATT_FORM_FIELD,
+    WattButtonComponent,
+    FormsModule,
+    ReactiveFormsModule,
+    WattInputDirective,
+    WattDatepickerComponent,
+    NgIf,
+    JsonPipe,
+  ],
+  standalone: true,
+  styles: [
+    `
+      watt-form-field {
+        margin-top: var(--watt-space-l);
+        margin-bottom: var(--watt-space-m);
+      }
+    `,
+  ],
+  template: `<watt-modal
+    #modal
+    [title]="title"
+    size="small"
+    closeLabel="Close modal"
+    [loading]="requestLoading"
+  >
+    <form [formGroup]="form">
+      <watt-form-field>
+        <watt-label>Receiver</watt-label>
+        <input wattInput required="true" type="text" formControlName="tin" [maxlength]="8" />
+        <watt-error *ngIf="form.controls.tin.errors"
+          >An 8-digit TIN/CVR number is required</watt-error
+        >
+      </watt-form-field>
+      <watt-form-field>
+        <watt-label>Period</watt-label>
+        <watt-datepicker required="true" formControlName="dateRange" [range]="true" />
+        <watt-error *ngIf="form.controls.dateRange.errors"> A date range is required </watt-error>
+      </watt-form-field>
+    </form>
+    <watt-modal-actions>
+      <watt-button variant="secondary" (click)="modal.close(false)">Close</watt-button>
+      <watt-button [disabled]="!form.valid" (click)="createAgreement()"
+        >Create transfer agreement</watt-button
+      >
+    </watt-modal-actions>
+  </watt-modal>`,
+})
+export class EoTransfersModalComponent {
+  @ViewChild(WattModalComponent) modal!: WattModalComponent;
+  @Input() title = '';
+
+  form = new FormGroup({
+    tin: new FormControl('', [Validators.minLength(8), Validators.pattern('^[0-9]*$')]),
+    dateRange: new FormControl<WattRange>({ start: '', end: '' }, [WattRangeValidators.required()]),
+  });
+  requestLoading = false;
+
+  constructor(
+    private service: EoTransfersService,
+    private store: EoTransfersStore,
+    private cd: ChangeDetectorRef
+  ) {}
+
+  open() {
+    this.modal.open();
+  }
+
+  createAgreement() {
+    if (!this.form.controls['dateRange'].value || !this.form.controls['tin'].value) return;
```

Comment:
Why not use `form.valid` ?

Reply:
Basically "because typescript" :) The fields can be null, and it does not like passing null values into "new Date"

Here is a picture of what it looks like. I avoid the errors by nullchecking in that if statement

![image](https://github.com/Energinet-DataHub/greenforce-frontend/assets/22421190/ed6ae3ef-553f-4d50-abba-18161d276b6a)

Reply:
I have put form.valid in now, just for good measure. But the issue here is mainly that typescript doesn't know i already checked for null outside the function :)